### DISCUSSION TITLE
Question: External Temperature

### DISCUSSION THREAD

**Discussion by theflu** [echavet/MitsubishiCN105ESPHome#193] [Q&A] [answered]
I see the recommended way to setup external temp is to use the home assistant api. In the event the sensor is no longer working what will happen? Will the unit revert to using its internal sensor or continue with the last known temperature? If it reverts how long does it take to do so?

My current setup uses actions in HA and sets the remote temp on every update HA receives. If the temp sensor hasn't updated in 10 min it will set the temp to 0 so it uses its internal sensor.

Also is the throttle needed? I currently send a new temp every ~10s.

**Comment by tmchow** [comment]
There is a timeout period for the remote sensor `remote_temperature_timeout` which controls this. It will revert to the internal temp if this timeout is passed.

**Comment by phidauex** [comment, marked as answer]
Hi, the two ways to update the temperature are to use an ESPHome "HomeAssistant Sensor" or to manually update the service with an HA Automation. The homeassistant sensor is the default and recommended right now because it is simpler and proving more reliable for most users.

If you have `remote_temperature_timeout` set, then yes, if the HA sensor stops sending updates, the unit will revert back to the internal temperature sensor after that timeout. I'm using 30 min right now, which seems to work well in my house. You could set it faster if your sensors update faster.

I'm the one who set the 30s throttle in the default settings - I was worried about forcing the unit to accept a new temperature every update cycle, which can be 2-4s, and I don't think there is any physical reason why you'd want it to accept updates that fast anyway - the goal is to slow down the response of the heatpump so it modulates better, so more filtering rather than less should get a better result. You could try a faster throttle if you wanted, however.

**Reply by theflu** [reply]
This is good to know, thank you. I wanted to switch to the api since it is easier to setup rather than managing automations for each unit.

I thought about adding temp sensor to the ESP board, but that would have been the same as using onboard sensor since I want to keep the ESP inside the unit.

**Reply by tmchow** [reply]
> I'm the one who set the 30s throttle in the default settings - I was worried about forcing the unit to accept a new temperature every update cycle, which can be 2-4s, and I don't think there is any physical reason why you'd want it to accept updates that fast anyway - the goal is to slow down the response of the heatpump so it modulates better, so more filtering rather than less should get a better result. You could try a faster throttle if you wanted, however.

Is this true for efficiency? My understanding is that the frequency of temperature updates wouldn't impact the efficiency of the heat pump. The temperature is just telling the heat pump with the current room temperature is then it decides to modulate. The fact that it's updating quickly shouldn't matter even if the temperature is staying the same. If the temperature is actually wildly changing in value say going up by 5° then you want to know that as fast as possible, don't you? Then the heat pump can adjust the temperature as needed.

In other words, are you saying that by slowing down the frequency of possible temperature update it makes it more efficient so the heat pump doesn't have to try to adjust things? That doesn't seem right or am I misunderstanding what you meant?

**Reply by phidauex** [reply]
> Is this true for efficiency? My understanding is that the frequency of temperature updates wouldn't impact the efficiency of the heat pump. The temperature is just telling the heat pump with the current room temperature is then it decides to modulate. The fact that it's updating quickly shouldn't matter even if the temperature is staying the same. If the temperature is actually wildly changing in value say going up by 5° then you want to know that as fast as possible, don't you? Then the heat pump can adjust the temperature as needed.
> 
> In other words, are you saying that by slowing down the frequency of possible temperature update it makes it more efficient so the heat pump doesn't have to try to adjust things? That doesn't seem right or am I misunderstanding what you meant?

I think what I mean is that heatpumps don't like rapid changes, because the efficiency is greatest at the lowest power outputs. By the same token, sensor systems have inherent noise, and your rooms are subject to swirling air currents, hot/cold spots, people walking by and breathing on it, etc.

The goal of your system isn't to maintain a perfect temperature at the exact sensor location and time stamp, but to maintain average temperature of the people in the room over time. You also want to keep the system running as efficiently as possible, meaning long low cycles. This would mean that you would actually be a bit motivated to allow the system to overshoot temperature sometimes, with a longer on cycle, and off cycle. Now, I don't think you need to micromanage this - the Mitsubishi systems have pretty smart PID tuning and handle themselves well, but I'm arguing that there is no advantage, and some potential disadvantage, to supplying rapidly changing temperature updates to the unit.

Most of my rooms have 2-3 sensors of different types, and I supply the average to the heatpump, which is nice for sensor redundancy, but it also acts as a bit of a moving average filter as air circulates in the room. Without multiple sensors, then I think the better choice would be to intentionally throttle and smooth the temp input to reduce the incidence of rapid peaks, then update the sensor at a reasonable rate, but in a world where you are shooting for, say 15 minute cycles at the shortest, updates faster than 1 minute don't seem to add value.

Now, my rooms are fairly large and my minisplits sized pretty precisely to the rooms, so none of them are heating up over setpoint in 5 minutes or anything, they increase temp fairly slowly and hold it, but if I had a small room that was heating up too fast, I'd be inclined to improve airflow to other rooms, and then allow the overshoot, going back to the long cycles = higher efficiency thing.

**Reply by tmchow** [reply]
I guess I'm still confused then as it relates to the actual temper measurement itself. Ignore for a second the fact that this is a remote temperature. Everything you said if taken at would also apply to the internal temperature so it would mean that the internal temperature shouldn't be read in real time either, right? 

Ultimately, I would think that temperature and accuracy should always be measured as quickly and often as possible, and it's the logic of the system that handles what to do with deviation temperature that should control whether or not it tries to heat or cool and chase that. 

For example, assume there's no remote temperature in place. If the internal temperature suddenly swung up 10° for five seconds, I wouldn't expect the Mitsubishi heat pump just suddenly chase that right away and go full blast immediately.

**Reply by phidauex** [reply]
My opinion on the matter isn't very strongly held - I don't think you'll break anything by updating faster (or rather, if it wasn't working you'd see it in communication failures in the logs), I just don't think it helps, and the SCADA guy in me just defaults to "what is the slowest update that will get a good result".

**Reply by theflu** [reply]
I do feel like the internal sensor would be read in real time or polled X seconds. I also feel like there is some sort of logic to increase the output if its calling for heat and the temp isn't going up. I would be nice to have an actually Mitsubishi cn105 thermostat and log the tx/rx to see how it interacts with the unit.

**Reply by tmchow** [reply]
In the case of my mini split here the internal temp of the ESP01 board ends up reading something insane like 85F for some reason. Not sure if it's because I have the module inside the casing of the minisplit

**Reply by theflu** [reply]
When I said internal sensor I meant the sensor built into the mini split itself.

**Reply by CopperSubstrate** [reply]
> In the case of my mini split here the internal temp of the ESP01 board ends up reading something insane like 85F for some reason. Not sure if it's because I have the module inside the casing of the minisplit

We have the same issue in most of our units, which is why I set up the remote sensors in the first place. 

In a room that 62 F the unit itself will report something like 75 F.
